app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                      $post= false;
+                      $site = false;
+                      if(strpos($val,'SITE-') !== false){
+                          $site = str_replace('SITE-', '', $val);
+                          return $dap->getBuilder()->andWhere("p.site_id = $site")->addGroupBy('p.id');
+                      } else{
+                          return $dap->getBuilder()->andWhere("p.id = $val")->addGroupBy('p.id');
+                      }
+                  })
+              )
+              ->setCallback(function($val, ObjectDataRow $row) {
+                  $site_url = $row->getCellValue('site_url');
+                  $target_post_id = $row->getCellValue('target_post_id');
+                  $post_link = !empty($target_post_id) ? $site_url . "?p=".$target_post_id : "";
+                  return '<a target="_blank" href="' . $post_link . '">' . $post_link . '</a>';
+              }),
+            (new FieldConfig)
+               ->setName('word_count')
+               ->setLabel('文字数')
+               ->setCallback(function($val){
+                   return $val === -1 ? '未更新' : $val;
+               }),
+              (new FieldConfig)
+                  ->setName('id_post_old')
+                  ->setLabel('元ID')
+                  ->setCallback(function($val){
+                      return $val === 0 ? '' : $val;
+                  }),
+            (new FieldConfig)
+              ->setName('is_delete')
+              ->setLabel('編集')
+              ->setCallback(function($val, ObjectDataRow $row) {
+                  $clonePost = new Post();
+                  $clonePost->id = $row->getCellValue('id');
+                  $clonePost->user_id = $row->getCellValue('user_id');
+                  $clonePost->is_delete = $row->getCellValue('is_delete');
+                  $clonePost->published_status = $row->getCellValue('published_status');
+
+                  $status = $row->getCellValue('published_status');
+                  $post_id = $row->getCellValue('id');
+                  $action_delete = $this->currentUser->is(User::ADMIN) ? 'Webapp\PostController@getDelete' : 'Webapp\PostController@getWriterDelete';
+                  $is_delete = '';
+                  if($val){
+                      $is_delete = '<span class="is-delete"></span>';
+                  }
+                  $url_pub_draft = '';
+
+                  $nextStatusButton = ($status == 1) ? 'Draft' : 'Publish';
+
+                  if($clonePost->{'can'.$nextStatusButton.'By'}($this->currentUser)) {
+                      $url_pub_draft = '<a href="'. action(($status==0 ? 'Webapp\PostController@getPublish' : 'Webapp\PostController@getDraft' ), $post_id) . '?hasFilter">'
+                          . '<span class="label'. ($status==0 ? ' label-success fa fa-check' : ' label-default fa fa-minus-circle' ) .'"><span class="tooltip action-top">'. ($status==0 ? '公開' : '下書きに戻す' ) .'</span></span>'
+                          . '</a> ';
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "</a> " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)